qiskit/pulse/macros.py

```diff
@@ -77,18 +113,67 @@ def measure(
                 "argument. For assistance, the instructions which are defined are: "
                 "{}".format(measure_name, inst_map.instructions)
             ) from ex
-        for time, inst in default_sched.instructions:
-            if inst.channel.index not in qubits:
-                continue
-            if qubit_mem_slots and isinstance(inst, instructions.Acquire):
-                if inst.channel.index in qubit_mem_slots:
-                    mem_slot = channels.MemorySlot(qubit_mem_slots[inst.channel.index])
-                else:
-                    mem_slot = channels.MemorySlot(unused_mem_slots.pop())
-                inst = instructions.Acquire(inst.duration, inst.channel, mem_slot=mem_slot)
-            # Measurement pulses should only be added if its qubit was measured by the user
-            schedule = schedule.insert(time, inst)
+        schedule = combine_schedules(qubits, schedule, default_sched, qubit_mem_slots, measure_group_qubits)
+    return schedule
+
+
+def measure_v2(
+    qubits: List[int],
+    backend=None,
+    target: Optional[Target] = None,
+    meas_map: Optional[Union[List[List[int]], Dict[int, List[int]]]] = None,
+    qubit_mem_slots: Optional[Dict[int, int]] = None,
+    measure_name: str = "measure",
+) -> Schedule:
+    """Return a schedule which measures the requested qubits according to the given
+    instruction mapping and measure map, or by using the defaults provided by the backend.
 
+    By default, the measurement results for each qubit are trivially mapped to the qubit
+    index. This behavior is overridden by qubit_mem_slots. For instance, to measure
+    qubit 0 into MemorySlot(1), qubit_mem_slots can be provided as {0: 1}.
+
+    Args:
+        qubits: List of qubits to be measured.
+        backend (Union[Backend, BaseBackend]): A backend instance, which contains
+            hardware-specific data required for scheduling.
+        target: The :class:`~.Target` representing the target backend, If None, defaults to the
+                  ``target`` of ``backend``.
+        meas_map: List of sets of qubits that must be measured together. If None, defaults to
+                  the ``meas_map`` of ``backend``.
+        qubit_mem_slots: Mapping of measured qubit index to classical bit index.
+        measure_name: Name of the measurement schedule.
+
+    Returns:
+        A measurement schedule corresponding to the inputs provided.
+
+    Raises:
+        PulseError: If both ``target`` and ``backend`` is None.
+    """
+    schedule = Schedule(name=f"Default measurement schedule for qubits {qubits}")
+    try:
+        target = target or backend.target
+        meas_map = meas_map or target.meas_map.get_qubit_groups(qubits)
+    except AttributeError as ex:
+        raise exceptions.PulseError(
+            "inst_map or meas_map, and backend cannot be None simultaneously"
+        ) from ex
+    if isinstance(meas_map, list):
+        meas_map = utils.format_meas_map(meas_map)
```

Comment:
Can you integrate this into `MeasureGrouping`?

Reply:
This function is used in only _measure_v1. So I think that it is no necessary to move into `MeasureGrouping`.